Install psycopg2 to the correct python virtual environment (#22613)
home-assistant/home-assistant.github.io: source/_integrations/recorder.markdown

pip3 install mysqlclient
```

> If Home Assistant is installed in a virtual environment, the additional libraries should be installed in the same virtual environment. If you installed Home Assistant Core to a Raspberry Pi following the [standard installation instructions](https://www.home-assistant.io/installation/raspberrypi) the following process should work.

Reviewer: Let's rewrite this a little bit, the important point is to activate the venv before running pip3.
Also, the link should be relative.
```suggestion
If Home Assistant is installed in a virtual environment, the additional libraries should be installed in the same virtual environment. If you installed Home Assistant Core to a Raspberry Pi following the [standard installation instructions](/installation/raspberrypi) the venv can be activated like this prior to installing either `pymysql` or `mysqlclient`:
```

```bash

> pip3 uninstall psycopg2
> sudo systemctl stop home-assistant@homeassistant.service
> sudo -u homeassistant -H -s
> source /srv/homeassistant/bin/activate
> pip3 install mysqlclient

Reviewer: Only the venv activation is relevant.
```suggestion
source /srv/homeassistant/bin/activate
pip3 install mysqlclient
deactivate
```

pip3 install psycopg2
```

If this doesn't work you may have to install the psycopg2 package as part of the same virtual environment that runs Home Assistant
```bash
pip3 uninstall psycopg2
sudo systemctl stop home-assistant@homeassistant.service
sudo -u homeassistant -H -s
source /srv/homeassistant/bin/activate
pip3 install psycopg2
```

Then hit Control-D to exit the virtual environment.

> 

Reviewer: I think it's better to document that if Home Assistant is installed in a virtual environment, the additional libraries should be installed in the same virtual environment. This is also the case for mysql / mariadb.

Author: I made changes based on your suggestion, but didn't notice that you had already made / suggested a change. I'm reasonably familiar with version control but not so much with HA or git. Apologies if I haven't done this quite correctly, and if you are able to fix it up please go ahead!

pip3 install psycopg2
```

> If Home Assistant is installed in a virtual environment, the additional libraries should be installed in the same virtual environment. If you installed Home Assistant Core to a Raspberry Pi following the [standard installation instructions](https://www.home-assistant.io/installation/raspberrypi) the following process should work.

Reviewer: Let's rewrite this a little bit, the important point is to activate the venv before running pip3.
Also, the link should be relative.
```suggestion
If Home Assistant is installed in a virtual environment, the additional libraries should be installed in the same virtual environment. If you installed Home Assistant Core to a Raspberry Pi following the [standard installation instructions](/installation/raspberrypi) the venv can be activated like this prior to installing `psycopg2`:
```

```bash

> pip3 uninstall psycopg2
> sudo systemctl stop home-assistant@homeassistant.service
> sudo -u homeassistant -H -s
> source /srv/homeassistant/bin/activate
> pip3 install psycopg2

Reviewer: ```suggestion
source /srv/homeassistant/bin/activate
pip3 install psycopg2
deactivate
```

```

> Then hit Control-D to exit the virtual environment.
> 
> 

Reviewer: ```suggestion
```
